packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItem.tsx

```diff
@@ -322,13 +339,24 @@ export const TreeItem = ({
     /* ref, tabindex and onFocus are used to set the React component focusabled and to set the focus to the corresponding DOM part */
     currentTreeItem = (
       <>
+        <div
+          className={`${state.partHovered === 'before' ? classes.treeItemHover : ''} ${classes.treeItemBefore}`}
+          onDrop={onDropBefore}
+          onDragEnter={() => handleMouseEnter('before')}
```

Comment:
I won't block the PR for that but this will update the state of this treeItem and thus re-render all of its children when we are dragging elements just to update some CSS. We may have to re-create that in pure CSS since it is very similar to another major performance issue identified on tree items.

You can change the style of an element because another one is `:hover` in CSS.

Reply:
See https://github.com/eclipse-sirius/sirius-web/pull/4139 for an attempt at this.